plugins/com.google.cloud.tools.eclipse.appengine.deploy.test/src/com/google/cloud/tools/eclipse/appengine/deploy/AppEngineDeployInfoTest.java

```diff
@@ -0,0 +1,85 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.core.runtime.CoreException;
+import org.junit.Test;
+import org.xml.sax.SAXException;
+
+public class AppEngineDeployInfoTest {
+
+  private static final String TEST_VERSION = "fooVersion";
+  private static final String TEST_ID = "fooId";
+
+  private static final String XML_SUFFIX = "</appengine-web-app>";
+  private static final String XML_DECLARATION = "<?xml version='1.0' encoding='utf-8'?>"
+                                              + "<appengine-web-app xmlns='http://appengine.google.com/ns/1.0'>";
+  private static final String XML_DECLARATION_WITH_INVALID_NS =
+      "<?xml version='1.0' encoding='utf-8'?><appengine-web-app xmlns='http://foo.bar.com/ns/42'>";
+  private static final String PROJECT_ID = "<application>" + TEST_ID + "</application>";
```

Comment:
I think it's worth adding some tests to include `<!-- comments -->` as they will occur in practice.